git-branchless-revset/src/builtins.rs

```diff
@@ -513,6 +564,37 @@ fn fn_current(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
     Ok(result.into_iter().collect::<CommitSet>())
 }
 
+#[instrument]
+fn fn_merges(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
+    let arg = eval0_or_1(ctx, name, args)?;
+    let arg = match arg {
+        Some(arg) => arg,
+        None => {
+            let visible_heads = ctx
+                .dag
+                .query_visible_heads()
+                .map_err(EvalError::OtherError)?;
+            let visible_commits = ctx.dag.query_ancestors(visible_heads.clone())?;
+            ctx.dag
+                .filter_visible_commits(visible_commits)
+                .map_err(EvalError::OtherError)?
+        }
+    };
```

Comment:
I chose to allow `merges()` to accept an arg because it seemed like a possible optimization. (Please correct me if I'm misunderstanding how revsets are evaluated.) My thinking is that `merges() & draft()` would have to calculate all visible merges and then intersect that w/ draft commits, while `merges(draft())` would be able to use the imput (ie draft commits) instead of all visible.

This differs from several other revset fns, though, that operate only on all visible commits w/o this sort of arg-optimization. Open to input, and happy to revert to a simpler only-visible commit version, if you prefer.

Reply:
FWIW this may not be a worthwhile optimization. I just played around in my work repo (11k commits including 2.8k merges) and got the following results:
```sh
❯ time git query 'merges() & date("after:1 year ago")' | wc -l
     571

________________________________________________________
Executed in    4.79 secs    fish           external
   usr time    3.87 secs    0.48 millis    3.87 secs
   sys time    4.06 secs    2.81 millis    4.06 secs


❯ time git query 'merges(date("after:1 year ago"))' | wc -l
     571

________________________________________________________
Executed in    4.07 secs    fish           external
   usr time    3.43 secs    0.45 millis    3.43 secs
   sys time    3.62 secs    2.77 millis    3.62 secs
```

So it's 10-15% faster in this case, but then a much smaller set of commits (`merges(draft())` vs `merges() & draft()`) yielded no speed up. So I guess I'm leaning towards simplifying this implementation and omitting this "optimization". What do you think?

Reply:
I would prefer to keep the zero-argument version only (and operate on all merges, not just draft commits) and try to optimize as per the above discussion on `contains`/`evaluate`.

Reply:
> keep the zero-argument version only

Done!